cob_cartesian_controller/src/cartesian_controller.cpp

```diff
@@ -29,430 +29,366 @@
 #include <algorithm>
 
 #include <ros/ros.h>
-#include "ros/package.h"
 #include <std_srvs/Empty.h>
 #include <cob_srvs/SetString.h>
-#include <kdl_conversions/kdl_msg.h>
 
 #include <cob_cartesian_controller/cartesian_controller.h>
-#include <cob_cartesian_controller/trajectory_interpolator/trajectory_interpolator.h>
 
 bool CartesianController::initialize()
 {
     ros::NodeHandle nh_private("~");
 
-    ///get params articulation Nodehandle
-    if(!nh_private.getParam("reference_frame", reference_frame_))
+    if (!nh_.getParam("chain_tip_link", chain_tip_link_))
     {
-        ROS_ERROR("Parameter 'reference_frame' not set");
+        ROS_ERROR("Parameter 'chain_tip_link' not set");
         return false;
     }
-
-    if(!nh_private.getParam("target_frame", target_frame_))
+    
+    if(!nh_.getParam("root_frame", root_frame_))
     {
-        ROS_ERROR("Parameter 'target_frame' not set");
+        ROS_ERROR("Parameter 'reference_frame' not set");
         return false;
     }
 
-    if (nh_private.hasParam("update_rate"))
-    {    nh_private.getParam("update_rate", update_rate_);    }
-    else
-    {    update_rate_ = 68.0;    }    //hz
-
+    if(!nh_.getParam("update_rate", update_rate_))
+    {
+        update_rate_ = 50.0;    //hz
+    }
 
-    /// Cartesian Nodehandle
-    if (!nh_.getParam("chain_tip_link", chain_tip_link_))
+    /// Private Nodehandle
+    if(!nh_private.getParam("target_frame", target_frame_))
     {
-        ROS_ERROR("Parameter 'chain_tip_link' not set");
-        return false;
+        ROS_WARN("Parameter 'target_frame' not set. Using default 'cartesian_target'");
+        target_frame_ = "cartesian_target";
     }
 
+
     ROS_WARN("Waiting for Services...");
     start_tracking_ = nh_.serviceClient<cob_srvs::SetString>("frame_tracker/start_tracking");
     stop_tracking_ = nh_.serviceClient<std_srvs::Empty>("frame_tracker/stop_tracking");
     start_tracking_.waitForExistence();
     stop_tracking_.waitForExistence();
+    tracking_ = false;
+
+    trajectory_interpolator_.reset(new TrajectoryInterpolator(root_frame_, update_rate_));
 
-    action_name_ = "cartesian_trajectory_action_";
+    action_name_ = "cartesian_trajectory_action";
     as_.reset(new SAS_CartesianControllerAction_t(nh_, action_name_, false));
     as_->registerGoalCallback(boost::bind(&CartesianController::goalCB, this));
```

Comment:
Similar to twist controller and obstacle distance: Write full name "Callback" in methods.